packages/ui/src/components/select.tsx

```diff
@@ -212,54 +213,76 @@ function Component<T extends SelectOption<ValueType>>(
                 placement="bottom-start"
                 className={{
                     root: dropdownRootStyles({
+                        empty: filteredOptions.length === 0,
                         className: className?.dropdownRoot,
                     }),
                 }}
                 ref={dropdownRef}
             >
-                <ul
-                    style={{ width: anchorEl?.parentElement?.clientWidth }}
-                    className={optionListStyles({
-                        className: className?.list,
-                    })}
-                >
-                    {filteredOptions.length === 0 && messages.noResults ? (
-                        <li
-                            className={optionStyles({
-                                noResults: true,
-                                className: className?.option,
-                            })}
-                        >
-                            {messages.noResults}
-                        </li>
-                    ) : (
-                        filteredOptions.map((option) => {
+                {filteredOptions.length === 0 ? (
+                    <div
+                        style={{
+                            width: anchorEl?.parentElement?.clientWidth,
+                        }}
+                        className={emptyOptionListStyles({
+                            className: className?.emptyList,
+                        })}
+                    >
+                        {messages.noResults}
+                    </div>
+                ) : (
+                    <AutoSizer disableWidth>
+                        {({ height }) => {
                             return (
-                                <li
-                                    className={optionStyles({
-                                        picked: value?.value === option.value,
-                                        className: className?.option,
+                                <FixedSizeList
+                                    height={height}
+                                    width={"auto"}
```

Comment:
Does it make sense to use width auto here and then later set it through inline styles?

Reply:
Fixed in 3ed97c23e9d9e240a6ab53d46e81013989c7c9f6.